cns/restserver/util.go

```diff
@@ -40,6 +40,31 @@ func (service *HTTPRestService) setNetworkInfo(networkName string, networkInfo *
 	service.state.Networks[networkName] = networkInfo
 }
 
+func (service *HTTPRestService) SetPnpIDMacaddressMapping(ctx context.Context) error {
+	p := platform.NewExecClient(nil)
+	VfMacAddressMapping, err := platform.FetchMacAddressPnpIDMapping(ctx, p)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch MACAddressPnpIDMapping")
+	}
+	logger.Printf("[DEBUG] Macaddress mapping: %d", len(VfMacAddressMapping))
+	service.PnpIDByMacAddress = VfMacAddressMapping
+	return nil
+}
+
+func (service *HTTPRestService) getPNPIDFromMacAddress(ctx context.Context, macAddress string) (string, error) {
+	if _, ok := service.PnpIDByMacAddress[macAddress]; !ok {
+		if err := service.SetPnpIDMacaddressMapping(ctx); err != nil {
```

Comment:
I'm not a fan of having `SetPnpIDMacaddressMapping` as a restservice function. `restservice` as it has already been overloaded with so many unrelated funcs already, plus all this function does is setting the service's PnpIDMacaddress map, can you make this a separated helper function that return the map instead?

Reply:
This function gets called from main function as well, Also, it saves response in service cache which is in rest server package.

Reply:
yes, we can still do that if this is a helper function instead. Since the only time in `SetPnpIDMacaddressMapping` referenced `service` is `service.PnpIDByMacAddress = VfMacAddressMapping`, so I think it would be a good idea to turn `SetPnpIDMacaddressMapping` into a separated helper function instead .i.e. `GetPnpIDMacaddressMapping` for example, and then we can set `service.PnpIDByMacAddress` to be the map returned by that func